src/main/java/com/thirdcc/webapp/service/impl/DebtServiceImpl.java

```diff
@@ -33,20 +35,101 @@ public DebtServiceImpl(DebtRepository debtRepository, DebtMapper debtMapper) {
+//        log.debug("Request to save Debt : {}", debtDTO);
+//        Debt debt = debtMapper.toEntity(debtDTO);
+//        debt = debtRepository.save(debt);
+//        return debtMapper.toDto(debt);
+//    }
+    
+//    /**
+//     * Update a debt.
+//     *
+//     * @param debtDTO the entity to update.
+//     * @return the persisted entity.
+//     */
+//    @Override
+//    public DebtDTO update(DebtDTO debtDTO) {
+//        log.debug("Request to save Debt : {}", debtDTO);
+//        Debt debt = debtRepository.findById(debtDTO.getId())
+//            .orElseThrow(() -> new BadRequestException("Event Checklist not found"));
+//        debt.setAmount(debtDTO.getAmount());
+//        debt.setEventAttendeeId(debtDTO.getEventAttendeeId());
+//        debt.setReceiptId(debtDTO.getEventAttendeeId());
+//        debt.setStatus(debtDTO.getStatus());
+//        return debtMapper.toDto(
+//            debtRepository.save(debt)
+//        );
+//    }
+
+//    /**
+//     * Get all the debts.
+//     *
+//     * @param pageable the pagination information.
+//     * @return the list of entities.
+//     */
+//    @Override
+//    @Transactional(readOnly = true)
+//    public Page<DebtDTO> findAll(Pageable pageable) {
+//        log.debug("Request to get all Debts");
+//        return debtRepository.findAll(pageable)
+//            .map(debtMapper::toDto);
+//    }
+
+
+//    /**
+//     * Get one debt by id.
+//     *
+//     * @param id the id of the entity.
+//     * @return the entity.
+//     */
+//    @Override
+//    @Transactional(readOnly = true)
+//    public Optional<DebtDTO> findOne(Long id) {
+//        log.debug("Request to get Debt : {}", id);
+//        return debtRepository.findById(id)
+//            .map(debtMapper::toDto);
+//    }
+
+//    /**
+//     * Delete the debt by id.
+//     *
+//     * @param id the id of the entity.
+//     */
+//    @Override
+//    public void delete(Long id) {
+//        log.debug("Request to delete Debt : {}", id);
+//        debtRepository.deleteById(id);
+//    }
+
     /**
-     * Save a debt.
+     * Update the debtStatus of the debt to "debtStatus" for "id" debt
      *
-     * @param debtDTO the entity to save.
-     * @return the persisted entity.
+     * @param id the id of the entity
+     * @param debtStatus the new debtStatus of the entity
+     * @return the entity.
      */
     @Override
-    public DebtDTO save(DebtDTO debtDTO) {
-        log.debug("Request to save Debt : {}", debtDTO);
-        Debt debt = debtMapper.toEntity(debtDTO);
-        debt = debtRepository.save(debt);
-        return debtMapper.toDto(debt);
+    public DebtDTO updateStatus(Long id, DebtStatus debtStatus) {
+        log.debug("Request to update status of debt: {}, to {}", id, debtStatus);
+        Debt debt = debtRepository.findById(id)
+            .orElseThrow(() -> new BadRequestException("Debt not exists: " + id));
+        if(!debt.getStatus().equals(DebtStatus.OPEN)){
```

Comment:
```
if(!debt.getStatus().equals(DebtStatus.OPEN))
```
`NullPointerException` will happen if dept.getStatus() is null (technically, our DB allow status column to be null)

better change it to be 
```
DebtStatus.OPEN.equals(debt.getStatus())
```

Reply:
ok sure